sdk/core/ts-http-runtime/scripts/azure-diff.ts

```diff
@@ -0,0 +1,182 @@
+  "src/util": "../core-util/src",
+
+  src: "../core-rest-pipeline/src",
+};
+
+const {
+  values: { update = false, "output-path": outputPath = "./review/azure-core-comparison.diff" },
+} = parseArgs({
+  options: {
+    update: {
+      type: "boolean",
+    },
+    "output-path": {
+      type: "string",
+      short: "o",
+    },
+  },
+});
+
+/**
+ * Recursively find all files in the directory
+ * @param dir - Directory to find files in
+ * @returns - an array containing paths to all files in the directory
+ */
+async function find(dir: string): Promise<string[]> {
+  const files = await fs.readdir(dir);
+  const allFiles: string[] = [];
+  for (const filename of files) {
+    const file = path.join(dir, filename);
+    const stat = await fs.stat(file);
+    if (stat.isDirectory()) {
+      allFiles.push(...(await find(file)));
+    } else {
+      allFiles.push(file);
+    }
+  }
+
+  return allFiles;
+}
+
+/**
+ * Returns a list of all unbranded source files and their azure counterparts if any
+ * @returns - list of tuples [unbranded file, azure counterpart]. The azure counterpart may be undefined if no counterpart can be found
+ */
+async function getFileMappings(): Promise<[string, string | undefined][]> {
+  const result: [string, string | undefined][] = [];
+  const files = await find("src");
+
+  for (const unbrandedLocation of files) {
+    const pathMatches = Object.keys(UNBRANDED_TO_AZURE_MAPPINGS).filter((x) =>
+      unbrandedLocation.startsWith(x),
+    );
+
+    let found = false;
+    for (const pathMatch of pathMatches) {
+      const azureLocation =
+        UNBRANDED_TO_AZURE_MAPPINGS[pathMatch] + unbrandedLocation.substring(pathMatch.length);
+
+      let exists: boolean;
+      try {
+        await fs.access(azureLocation);
+        exists = true;
+      } catch {
+        exists = false;
+      }
+
+      if (exists) {
+        result.push([unbrandedLocation, azureLocation]);
+        found = true;
+        break;
+      }
+    }
+
+    if (!found) {
+      result.push([unbrandedLocation, undefined]);
+    }
+  }
+
+  return result;
+}
+
+export interface RunResult {
+  exitCode?: number;
+  signal?: NodeJS.Signals;
+  stdout: string;
+}
+
+/**
+ * Runs the given CLI command and captures the output
+ */
+async function run(command: string, ...args: string[]): Promise<RunResult> {
+  let stdout = "";
+
+  const proc = spawn(command, args);
+
+  return new Promise((resolve, reject) => {
+    proc.on("error", reject);
+    proc.on("exit", (exitCode, signal) => {
+      resolve({
+        exitCode: exitCode ?? undefined,
+        signal: signal ?? undefined,
+        stdout,
+      });
+    });
+    proc.stdout.on("data", (data) => {
+      stdout += data.toString();
+    });
+  });
+}
+
+async function calculatePatchFileContents(): Promise<string> {
+  const mappings = await getFileMappings();
+
+  const patches: string[] = [];
+  for (const [unbrandedFile, azureFile = "/dev/null"] of mappings) {
```

Comment:
Is this meant to be "if azureFile has a diff with unbrandedFile, or if aszureFile exists but unbrandedFile does not" then show a diff? 

I think it's a little hard to parse if so, maybe a conditional will be easier to understand? If not and you like the way the code paths consolidate then maybe a comment explaining the purpose of defaulting to /dev/null.

Another Q from a WSL user - does this work on Windows?

Reply:
Yeah I get this might be a bit confusing. The new approach is a bit simpler and is hopefully easier to grok.

FWIW though using `/dev/null` here does work on Windows -- Git on Windows runs in a unix-like environment of some kind which does have access to `/dev/null`. The intent was to show files that were created in the unbranded package even if there isn't a corresponding file in the Azure package -- doing a diff with /dev/null makes the diff show the whole file being added.